Lombiq.Hosting.Tenants.FeaturesGuard/Handlers/FeaturesEventHandler.cs

```diff
@@ -54,56 +51,63 @@ async Task IFeatureEventHandler.DisabledAsync(IFeatureInfo feature)
     /// Enables conditional features (key) if one of their corresponding condition features (value) was enabled.
     /// </summary>
     /// <param name="featureInfo">The feature that was just enabled.</param>
-    public async Task EnableConditionallyEnabledFeaturesAsync(IFeatureInfo featureInfo)
+    public Task EnableConditionallyEnabledFeaturesAsync(IFeatureInfo featureInfo)
     {
-        if (_shellSettings.IsDefaultShell() ||
-            _conditionallyEnabledFeaturesOptions.Value.EnableFeatureIfOtherFeatureIsEnabled is not { } conditionallyEnabledFeatures)
+        // We are using a Deferred Task so we can avoid recursive executions upon tenant setup.
+        ShellScope.AddDeferredTask(async scope =>
         {
-            return;
-        }
+            if (_shellSettings.IsDefaultShell() || !IsConditionallyEnabledFeaturesOptionsPopulated(out var conditionallyEnabledFeatures))
+            {
+                return;
+            }
 
-        var allConditionFeatureIds = new List<string>();
-        foreach (var conditionFeatureIds in conditionallyEnabledFeatures.Values)
-        {
-            allConditionFeatureIds.AddRange(conditionFeatureIds);
-        }
+            var allConditionFeatureIds = GetAllConditionFeatureIds(conditionallyEnabledFeatures, featureInfo);
```

Comment:
Why are we not just enabling all features from `allConditionFeatureIds` after this point? I don't get why we need all the other logic below (except for not trying to enable features that are already enabled, but that's something that should be handled by `IShellFeaturesManager` anyway).

Reply:
Because the **Condition** features are the features that define whether the **Conditional** feature should be enabled or not. The whole Handler logic is to determine which **Conditional** feature to enable depending on the `featureInfo` if it is a condition feature.

Reply:
I see, thanks for explaining.